dwave_networkx/generators/pegasus.py

```diff
@@ -31,9 +31,19 @@
            ]
 
 
+def _add_compatible_edges(G, edge_list):
+    # Slow when edge_list is large, but clear (non-defaulted behaviour, so fine):
+    if edge_list is not None:
+        if not all([G.has_edge(*e) for e in edge_list]):
+            raise ValueError("edge_list contains edges incompatible with a "
+                             "fully yielded graph of the requested topology")
+        if len(edge_list) < G.number_of_edges():
```

Comment:
I missed this -- unfortunately, if `edge_list` has duplicates you run into issues here.  Repeat this comment 3 times (or refactor this function into a `generators/common.py` file).

Reply:
Given that this is a check on argument consistency, might be better just to document that it will check for duplicates and reject cases with duplicates. Seems like an edge case that will almost always catch a user error.

Generators include lots of common code, this being the worst case. I agree would be a good idea to create a new common file.